docs(sources/cloudwatch): validated AWS CloudWatch via OTel Collector to GlassFlow pattern (ETL-1163) (#769)
glassflow/clickhouse-etl: docs/app/sources/cloudwatch/page.mdx

    endpoint: http://<glassflow-otlp-receiver>:4318
    compression: none
    headers:
      x-glassflow-pipeline-id: <your-logs-pipeline-id>

  otlp_http/glassflow_metrics:
    endpoint: http://<glassflow-otlp-receiver>:4318
    compression: none
    headers:
      x-glassflow-pipeline-id: <your-metrics-pipeline-id>

service:
  pipelines:
    logs:
      receivers: [awscloudwatch/logs]
      processors: [batch]
      exporters: [otlp_http/glassflow_logs]
    metrics:
      receivers: [awscloudwatch/metrics]
      processors: [batch]
      exporters: [otlp_http/glassflow_metrics]
```

<Callout type="warning">
**Disable gzip compression** on the `otlp_http` exporter (`compression: none`). The GlassFlow OTLP receiver does not decompress gzip-encoded payloads and will return `400 Bad Request` if you leave the exporter's default `gzip` compression enabled.
</Callout>

One OpenTelemetry Collector instance can drive both signals in parallel. Logs and metrics need their own GlassFlow pipelines (`otlp.logs` source and `otlp.metrics` source respectively), so route each signal to its own exporter with the matching `x-glassflow-pipeline-id`.

### Minimum IAM permissions

> The credentials used by the Collector need read-only access to the CloudWatch APIs the receiver calls. Scope `Resource` to the specific log group ARNs and metric namespaces in production; the wildcard below is suitable for validation only.

Reviewer: ```suggestion
The credentials used by the Collector require read-only access to the specific CloudWatch APIs called by the receiver. While a wildcard is fine for initial validation, you should restrict the 'Resource' scope to specific log group ARNs and metric namespaces in production.
```

| `attributes` | `{'Namespace': 'AWS/EC2', 'MetricName': 'CPUUtilization', 'Dimensions': '{...}'}` |

The receiver prepends `amazonaws.com/<Namespace>/` to the metric name per the [CloudWatch Metric Streams OpenTelemetry 1.0 format](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-metric-streams-formats-opentelemetry-100.html).

Numeric metric values land in different schema fields depending on the receiver's `stats` configuration:

- **No `stats` (default).** Each datapoint is an OTel Summary built from `Sum`, `SampleCount`, `Minimum`, `Maximum`. Map the GlassFlow OTLP fields `sum`, `count`, `min`, `max` in your sink to capture the full statistical picture.
- **Explicit `stats: [Average]` (or any single stat).** Each datapoint is a Gauge with one value. Map the GlassFlow OTLP field `value_double` in your sink.

For full OTLP source field reference, see [OTLP Schema Reference](/sources/otlp/schema-reference). For the GlassFlow pipeline shape that writes these records to ClickHouse, see [OTLP Examples](/sources/otlp/examples).

## Latency

`awscloudwatchreceiver` is pull-based. Three timings stack:

> - **CloudWatch ingestion lag.** AWS itself takes 1 to 10 minutes to make a published log event or metric datapoint queryable. The receiver's default `delay: 10m` on metrics accounts for this; reduce it only after confirming CloudWatch's own lag for your region and metric source is consistently lower.

Reviewer: ```suggestion
- **CloudWatch ingestion lag.** AWS itself takes anywhere between 1 and 10 minutes to make a published log event or metric datapoint queryable. The receiver's default `delay: 10m` on metrics accounts for this; reduce it only after confirming CloudWatch's own lag for your region and metric source is consistently lower.
```

- **Logs `poll_interval`.** Default `1m`. Each poll fetches new events since the last poll.
- **Metrics `collection_interval`.** Default `5m`, lowered to `1m` in the example above for the smallest reasonable batch size.

> End-to-end first-event latency is typically 2 to 12 minutes depending on the AWS source and the receiver settings. If you need sub-minute latency, see the push alternative below.

Reviewer: ```suggestion
End-to-end first-event latency is typically 2 to 12 minutes, depending on the AWS source and the receiver settings. If you need sub-minute latency, see the push alternative below.
```

## Push alternative: Kinesis Data Firehose

For sub-minute latency on metrics and logs, AWS supports two push paths that can avoid the receiver's poll lag:

- **Metrics:** [CloudWatch Metric Streams](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Metric-Streams.html) to Firehose's HTTP endpoint, fronted by the [`awsfirehosereceiver`](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/awsfirehosereceiver) in alpha.
- **Logs:** CloudWatch Logs subscription filter to Firehose with the same `awsfirehosereceiver` extension.

> This trades the simple pull config for an HTTPS endpoint, IAM for Firehose, and the Firehose record envelope. The advanced path is out of scope for this page; reach out via Slack if you want to validate it end-to-end.

Reviewer: ```suggestion
This trades the simple pull config for an HTTPS endpoint, IAM for Firehose, and the Firehose record envelope. The advanced path is out of scope for this page; reach out via Slack or [get in touch](https://www.glassflow.dev/contact) with our team if you want to validate it end-to-end.
```
